botorch/models/fully_bayesian.py

```diff
@@ -534,11 +533,40 @@ def posterior(
         """
         self._check_if_fitted()
         posterior = super().posterior(
-            X=X,
+            X=X.unsqueeze(MCMC_DIM),
             output_indices=output_indices,
             observation_noise=observation_noise,
             posterior_transform=posterior_transform,
             **kwargs,
         )
         posterior = GaussianMixturePosterior(distribution=posterior.distribution)
         return posterior
+
+    def condition_on_observations(
+        self, X: Tensor, Y: Tensor, **kwargs: Any
+    ) -> BatchedMultiOutputGPyTorchModel:
+        """Conditions on additional observations for a Fully Bayesian model (either
+        identical across models or unique per-model).
+
+        Args:
+            X: (Tensor): A `(batch_shape) x num_samples x d`-dim Tensor, where `d` is
+                the dimension of the feature space and `batch_shape` is the number of
+                sampled models.
+            Y (Tensor): A `(batch_shape) x num_samples x 1`-dim Tensor, where `d` is
+                the dimension of the feature space and `batch_shape` is the number of
+                sampled models.
+
+        Returns:
+            BatchedMultiOutputGPyTorchModel: _description_
+        """
+        if X.ndim < 3 or Y.ndim < 3:
+            # The can either be thrown here or in GPyTorch, when the inference of the
+            # batch dimension fails since the training data by default does not have
+            # a batch shape.
+            raise ValueError(
+                "Conditioning in fully Bayesian models must contain a batch dimension."
+                "Add a batch dimension (the leading dim) with length matching the "
+                "number of hyperparameter sets to the conditioned data."
```

Comment:
Is the number of hyper-parameter sets `self.num_mcmc_samples` or `self.batch_shape` (the two are the same, unless the model is already batched / fantasized)? If I were to attempt fantasizing from an already fantasized model, would this still hold? It'd be good to print out the expected shape as part of the error message.

Reply:
The number of hyperparameter sets is `num_mcmc_samples / thinning` (neither is an attribute, so no attribute indicating the number of models exists for the FBGB). `self.batch_shape` is added when conditioning on observations, since `cond_X`, `cond_Y` have a `batch_shape.` I have added a test for this now, too.

I presume that the logic behind not having a `batch_shape` by default to not carry `batch_shape` sets of training data by default, and so now it is only added when it is necessary.

Reply:
Had a bit of an incorrect rushed comment before, which I deleted. =)